universal-application-tool-0.0.1/app/services/applicant/question/AddressQuestion.java

```diff
@@ -239,14 +239,17 @@ public boolean isAnswered() {
 
   @Override
   public String getAnswerString() {
-    String line1 = getStreetValue().orElse("");
-    String[] parts = {
-      getCityValue().orElse(""), getStateValue().orElse(""), getZipValue().orElse("")
-    };
-    String line2 =
-        Arrays.stream(parts).filter(part -> part.length() > 0).collect(Collectors.joining(", "));
-
-    String[] ret = {line1, line2};
-    return Arrays.stream(ret).filter(part -> part.length() > 0).collect(Collectors.joining("\n"));
+    String displayLine1 = getStreetValue().orElse("");
+    String displayLine2 = getLine2Value().orElse("");
+
+    String cityDisplayString = getCityValue().isPresent() ? getCityValue().get() + "," : "";
+    String displayLine3 =
+        String.format(
+            "%s %s %s", cityDisplayString, getStateValue().orElse(""), getZipValue().orElse(""));
```

Comment:
A. If `city` is blank wouldn't this have a leading space?
B. If `state` is blank wouldn't this have two spaces between `city` and `zip`?
C. If `zip` is blank wouldn't this have a trailing space?

A and C are easily solved by `.trim()`
For B you could use an if statement or ternary?

```
String stateDisplayString = getStateValue.orElse("");
String displayLine3 = stateDisplayString.isEmpty() ? : 
     String.format( "%s %s", cityDisplayString, getZipValue().orElse("")).trim() :
     String.format( "%s %s %s", cityDisplayString, stateDisplayString, getZipValue().orElse("")).trim();
```

Reply:
I tried to catch the leading/trailing spaces with a `trim` in the return statement, but didn't account for the double space between the city and zip! Nice catch.